src/DwitTech.AccountService.Data/repository/UserRepository.cs

```diff
@@ -108,6 +109,30 @@ private IQueryable<User> GetActiveUsers()
         {
             return _dbContext.Users.Where(x => x.Status != UserStatus.Deleted);
         }
+
+        public bool FindPasswordResetToken(string token)
+        {
+            var tokenExists = _dbContext.ValidationCodes.Any(x => x.Code == token);
+            return tokenExists;
+        }
+
+        public async Task<int> GetUserIdByPasswordResetToken(string token)
+        {
+            var result = await _dbContext.ValidationCodes.Where(x => x.Code == token).FirstOrDefaultAsync();
+            return result.UserId;
+        }
+
+        public async Task UpdateUserLoginsPassword(UserLogin userLogin)
+        {
+            _dbContext.UserLogins.Update(userLogin);
+            await _dbContext.SaveChangesAsync();
+        }
+
+        public async Task<UserLogin> GetUserLoginsByUserId(int userId)
+        {
+            var result = await _dbContext.UserLogins.Where(x => x.UserId == userId).FirstOrDefaultAsync();
```

Comment:
If I remember correctly, boss asked us not to do this because it commits the password to memory, which increases security risks.